assets/js/modules/analytics/components/dashboard/GA4DashboardWidgetSwitcher.js

```diff
@@ -36,6 +36,11 @@ export default function GA4DashboardWidgetSwitcher( {
 	const isGA4DashboardView = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS ).isGA4DashboardView()
 	);
+	const { WidgetNull } = widgetProps;
+
+	if ( isGA4DashboardView === undefined ) {
+		return <WidgetNull />;
+	}
```

Comment:
This change is required. Otherwise, the `UA` component will be rendered initially. That will cause network failure if the `UA` module isn't connected.

Reply:
Is rendering `<WidgetNull />` appropriate here? That's usually something we reserve for the widget being "empty" because there's no content to show, signifying to the widget areas/contexts above that nothing will be rendered here and other widgets can be re-flowed to take its place.

But that isn't _really_ the case here; we temporarily want to render nothing to prevent the network request, but if this widget is rendered at all we expect SOMETHING to be here. If `isGA4DashboardView` took awhile to resolve, that might even cause widget reflow.

In practice this isn't a huge deal, but it seems like it might be better to render `null` or some kind of loading screen if possible?

But maybe the best case is `<WidgetNull />`, what do you think given what I mentioned here?

Reply:
Thanks for your suggestions, @tofumatt. My idea was this component would render the widgets eventually. Render the `WidgetNull` while the resolver resolves. I don't strongly prefer whether to render `null` or `WidgetNull` here.

Reply:
We do render `WidgetNull` when the data is loading. Please check the condition for `undefined`:
https://github.com/google/site-kit-wp/blob/f2debd81a7dbcfe9a3dc62c954d7ccd18e9c0582/assets/js/components/KeyMetrics/KeyMetricsSetupCTAWidget.js#L47-L55